test/javascripts/acceptance/chat-live-pane-test.js

```diff
@@ -170,3 +170,63 @@ acceptance(
     });
   }
 );
+
+acceptance(
+  "Discourse Chat - Chat live pane - handling 429 errors",
+  function (needs) {
+    needs.user({
+      username: "eviltrout",
+      id: 1,
+      can_chat: true,
+      has_chat_enabled: true,
+    });
+    needs.settings({
+      chat_enabled: true,
+    });
+
+    needs.pretender((server, helper) => {
+      server.get("/chat/:chatChannelId/messages.json", () => {
+        return helper.response(429, {
+          errors: [
+            "You’ve performed this action too many times. Please wait 10 seconds before trying again.",
+          ],
+          error_type: "rate_limit",
+          extras: { wait_seconds: 10 },
+        });
+      });
+
+      server.get("/chat/chat_channels.json", () =>
+        helper.response({
+          public_channels: [
+            {
+              id: 1,
+              title: "something",
+              current_user_membership: { following: true },
+            },
+          ],
+          direct_message_channels: [],
+        })
+      );
+
+      server.get("/chat/chat_channels/:chatChannelId", () =>
+        helper.response({ id: 1, title: "something" })
+      );
+
+      server.post("/chat/drafts", () => {
+        return helper.response([]);
+      });
+
+      server.post("/chat/:chatChannelId.json", () => {
+        return helper.response({ success: "OK" });
+      });
+    });
+
+    test("Handles 429 errors by displaying an alert", async function (assert) {
+      await visit("/chat/channel/1/cat");
+
+      assert.ok(exists(`.bootbox`), "We displayed a 429 error");
+
+      await click(".bootbox span.d-button-label");
```

Comment:
Do we need this last click action? We're not asserting anything after it...

Reply:
We should clean things up after the test finishes. The modal doesn't close itself.